rpc/src/commands/chains/blocks/get_blocks_in_chain.rs

```diff
@@ -1,16 +1,88 @@
+use super::block_responses::BlocksInChainResponse;
 use crate::commands::RpcClientCommand;
-use crate::responses::chains::blocks::block_ids_in_chain::BlocksInChainResponse;
 use crate::types::Chain;
+use chrono::NaiveDateTime;
+use querystring;
 
+type BlockHash = String;
+
+#[derive(Debug)]
+/// Command for the [`/chains/{chain_id}/blocks` endpoint](https://tezos.gitlab.io/shell/rpc.html#get-chains-chain-id-blocks).
+///
+/// Requires a valid [`chain_id`](Chain) to form the URL string
+///
+/// Can optionally hold extra query parameters with
+/// [an explicit constructor](Self::with_explicit_params),
+/// or default to sending no args with [the default constructor](Self::with_default_params)
 pub struct GetBlocksInChain {
     pub chain_id: Chain,
+    params: Option<GetBlocksInChainParameters>,
+}
+
+#[derive(Debug)]
+struct GetBlocksInChainParameters {
+    length: Option<u32>,
+    head: Option<BlockHash>,
+    min_date: Option<NaiveDateTime>,
+}
+
+impl GetBlocksInChainParameters {
+    fn to_url_string(&self) -> String {
+        let mut query_pairs = Vec::new();
+
+        if let Some(length) = &self.length {
+            query_pairs.push(("length", length.to_string()));
+        }
+        if let Some(head) = &self.head {
+            query_pairs.push(("head", head.to_string()));
+        }
+        if let Some(min_date) = &self.min_date {
+            query_pairs.push(("min_date", min_date.timestamp().to_string()));
+        }
+
+        let query_params = query_pairs.iter().map(|x| (x.0, x.1.as_str())).collect();
+
+        querystring::stringify(query_params)
```

Comment:
`querystring::stringify` takes a `Vec<&'a str, &'a str>`, so I couldn't actually borrow the `str`'s for long enough in the `if let`s, so this was a cute little solution to turn the `Vec<&str, String>` into a `Vec<&'a str, &'a str>`.

Shoutout to compile-time checked tuples for the `x.0, x.1` syntax (no `unwrap`ing!)